delta-api/src/main/java/io/cdap/delta/api/DMLEvent.java

```diff
@@ -182,8 +186,53 @@ public Builder setRowId(String rowId) {
     }
 
     public DMLEvent build() {
-      return new DMLEvent(offset, database, new DMLOperation(table, operationType, ingestTimestampMillis), row,
-                          previousRow, transactionId, ingestTimestampMillis, sourceTimestampMillis, isSnapshot, rowId);
+      int sizeInBytes = (operationType == DMLOperation.Type.INSERT ||
+        operationType == DMLOperation.Type.UPDATE) ? computeSizeInBytes(row) : 0;
+
+      return new DMLEvent(offset, database, new DMLOperation(table, operationType, ingestTimestampMillis, sizeInBytes),
+                          row, previousRow, transactionId, ingestTimestampMillis, sourceTimestampMillis, isSnapshot,
+                          rowId);
+    }
+
+    private int computeSizeInBytes(StructuredRecord row) {
+      List<Schema.Field> fields = row.getSchema().getFields();
+      if (fields == null) {
+        return 0;
+      }
+      int sizeInBytes = 0;
+      int numBooleans = 0;
+      for (Schema.Field field : fields) {
+        Schema fieldSchema = field.getSchema();
+        fieldSchema = fieldSchema.isNullable() ? fieldSchema.getNonNullable() : fieldSchema;
+        if (row.get(field.getName()) == null) {
+          continue;
+        }
+        switch (fieldSchema.getType()) {
+          case BOOLEAN:
+            numBooleans++;
+            break;
+          case INT:
+          case FLOAT:
+          case ENUM:
+            sizeInBytes += 4;
+            break;
+          case LONG:
+          case DOUBLE:
+            sizeInBytes += 8;
+            break;
+          case BYTES:
+            byte[] bytes = row.get(field.getName());
+            sizeInBytes += bytes.length;
+            break;
+          case STRING:
+            String strValue = row.get(field.getName());
+            sizeInBytes += strValue.getBytes(StandardCharsets.UTF_8).length;
+            break;
+          default:
+            break;
+        }
+      }
+      return sizeInBytes + numBooleans / 8;
```

Comment:
although booleans can be represented by bit but multiple booleans won't be merged into one byte. one boolean still occupies one byte.

Reply:
+1, should be able to just do sizeInBytes++ for the boolean case

Reply:
done.